### DISCUSSION TITLE
Speed auto zoom (reverse)

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#88] [Q&A] [answered]
Speed auto-zoom is working fine now - for cars. I have temporarily tested it between 12 and 18.

I will nevertheless not use it (set back to 16,5 to 18) since I would need it exactly the other way round, strong zooming in when driving fast and zooming out when driving slow. When driving fast (mostly) I only want to see 1-2 (max. 3) next curves, nothing more. When driving slower (sometimes) I would like to see more route.

Would it be much effort to make the zooming exactly the opposite - optionally by a separate setting?

_Originally posted by @DolomitesTom in https://github.com/devemux86/cruiser/discussions/63#discussioncomment-5352767_

**Comment by devemux86** [comment]
With the new [speed auto zoom](https://github.com/devemux86/cruiser/discussions/63), it was easy to implement.

@DolomitesTom sorry for the [delay](https://forum.kurviger.de/t/reverse-speed-based-auto-zoom/4434)! :slightly_smiling_face: 

It will be available in the next Cruiser update.

<img src="https://user-images.githubusercontent.com/3484020/226109054-8ac07857-77e1-4f6e-abbf-003e3e80d397.jpg" width="50%"/>

**Reply by DolomitesTom** [reply]
Overwhealming. Wow.

**Comment by devemux86** [comment, marked as answer]
The reverse speed auto zoom is available in [Cruiser 3.0.13](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
"Settings | Navigation | Auto zoom | Auto zoom mode | Speed (reverse)" or search settings for "auto zoom".

Please let me know how you see it.

**Reply by DolomitesTom** [reply]
will test it this afternoon when the rain has gone

**Comment by DolomitesTom** [comment]
It works fantastic. First I thought there was a bug, but BMW connector was the bad guy. It opens automatically in the background when the bluetooth connection is established with the bike. The result is that sometimes Cruiser "loses" the GPS position. The arrow jumps completely off the route and back to the route within half a second. This happens pretty often then. The reason is probably because BMW connector makes somehow exclusive accesses to the gps position? Or is it a performance problem then? I don't know. After deleting the connection from the BMW and take away location access from the BMW connector app it works perfect as it should.

**Reply by devemux86** [reply]
> It works fantastic.

Thanks for the feedback, that's great!

> The reason is probably because BMW connector makes somehow exclusive accesses to the gps position?

Some apps can even set mock locations on GPS, like the mock location apps on Google Play.
More apps using the GPS should work, if all of them only read the GPS sensor, nothing more.

For example the [GPSTest](https://play.google.com/store/apps/details?id=com.android.gpstest) app is often used together with other apps?

What is selected in app "Settings | Location | Location service"?

**Reply by DolomitesTom** [reply]
> What` is selected in app "Settings | Location | Location service"?

Android

**Reply by devemux86** [reply]
Does it happen for the first time with this version?
I do not remember changing anything in the location libraries.

Can you also test with the "Google" location service?

**Comment by DolomitesTom** [comment]
> Does it happen for the first time with this version?

yes, but I also used the BMW connector for the first time since a long while, so I am sure this was the reason.

> I do not remember changing anything in the location libraries.
Can you also test with the "Google" location service?

could do so, but I like Android more because it is more precise.

No problem I don't like the bmw connector app because it also penetratingly steals the Bluetooth channel from my helmet intercom.
The only advantage is the built-in traffic information. But this is bought with all kinds of other disadvantages.

**Comment by DolomitesTom** [comment]
The speed reverse auto-zoom mode is superb. No other navigation app offers such a genius function. I am thrilled.

**Reply by devemux86** [reply]
@DolomitesTom thanks for the idea and the tests! :+1:

**Comment by DolomitesTom** [comment]
Have I mentioned how fantastic I find the Auto-Zoom speed reverse feature? 😆... There are also no more jumps in it. This was probably a lack of GPS precision at the time, which happens sometimes.

**Reply by devemux86** [reply]
New speed auto zoom (normal / reverse) use similar "lazy" sampling to avoid frequent zoom changes.
